examples/assets/scripts/parsers/hdr-texture.js

```diff
@@ -0,0 +1,222 @@
+function Reader(data) {
+    this.view = new DataView(data);
+    this.offset = 0;
+}
+
+Object.assign(Reader.prototype, {
+    readLine: function () {
+        var view = this.view;
+        var result = "";
+        while (true) {
+            if (this.offset >= view.byteLength) {
+                break;
+            }
+
+            var c = String.fromCharCode(this.readUint8());
+            if (c === '\n') {
+                break;
+            }
+            result += c;
+        }
+        return result;
+    },
+
+    readUint8: function () {
+        return this.view.getUint8(this.offset++);
+    },
+
+    readUint8s: function (result) {
+        var view = this.view;
+        for (var i = 0; i < result.length; ++i) {
+            result[i] = view.getUint8(this.offset++);
+        }
+    },
+
+    peekUint8s: function (result) {
+        var view = this.view;
+        for (var i = 0; i < result.length; ++i) {
+            result[i] = view.getUint8(this.offset + i);
+        }
+    }
+});
+
+/**
+ * @class
+ * @name pc.HdrParser
+ * @implements {pc.TextureParser}
+ * @classdesc Texture parser for hdr files.
+ */
+var HdrParser = function (registry, retryRequests) {
```

Comment:
Just as a matter of consistency, I'd stick to a regular function declaration rather than a function expression here.